.github/.check_scopes.lua

```diff
@@ -0,0 +1,49 @@
+-- Copyright Cartesi Pte. Ltd.
+-- 
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+-- this file except in compliance with the License. You may obtain a copy of the
+-- License at http://www.apache.org/licenses/LICENSE-2.0
+-- 
+-- Unless required by applicable law or agreed to in writing, software distributed
+-- under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+-- CONDITIONS OF ANY KIND, either express or implied. See the License for the
+-- specific language governing permissions and limitations under the License.
+
+function extractScopesFromFile(filename)
+  local scopes = {}
+  local file = io.open(filename)
+  if not file then
+    error("Failed to open file: " .. filename)
+  end
+  for line in file:lines() do
+    local scope = line:match("%S+")
+    if scope then
+      scopes[scope] = true
+    end
+  end
+  file:close()
```

Comment:
We could omit this line, since, according to the [Lua 5.4 documentation](https://www.lua.org/manual/5.4/manual.html#pdf-file:close)...

> [...] files are automatically closed when their handles are garbage collected [...]

I don't think this is too bad, since we are only opening one file.

Reply:
It is good practice to close the file since we don't know when the garbage collector will run.

Reply:
Yes, this is fair.